crates/sui-graphql-rpc/src/types/object.rs

```diff
@@ -161,7 +164,25 @@ pub(crate) enum ObjectVersionKey {
     Historical(u64), // version
 }
 
-pub(crate) type Cursor = cursor::BcsCursor<Vec<u8>>;
+pub(crate) enum HistoricalObjectPaginationResult<I, T>
+where
+    I: Iterator<Item = T>,
+{
+    Error(HistoricalObjectPaginationError),
+    Success(bool, bool, I),
+}
```

Comment:
I can't figure out how to get execute_repeatable to return Result<T, MyCustomError>, since in the process of running the query it'll convert errors to IndexerError, and then to InternalError. Hence this type as a workaround